mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,660 @@
+        # number of consecutive frames
+        self.clip_len = clip_len
+        # number of steps (sparse sampling for efficiency of io)
+        self.frame_interval = frame_interval
+
+        # test mode or not
+        self.filter_gt = filter_gt
+        self.test_mode = test_mode
+
+        if self.verbose:
+            print("""
+            SSNDataset: proposal file {prop_file} parsed.
+
+            There are {pnum} usable proposals from {vnum} videos.
+            {fnum} foreground proposals
+            {inum} incomplete proposals
+            {bnum} background proposals
+
+            Sample config:
+            FG/BG/INC: {fr}/{br}/{ir}
+            Video Centric: {vc}
+
+            Regression Stats:
+            Location: mean {stats[0][0]:.05f} std {stats[1][0]:.05f}
+            Duration: mean {stats[0][1]:.05f} std {stats[1][1]:.05f}
+            """.format(
+                prop_file=self.proposal_file,
+                pnum=len(self.fg_pool) + len(self.bg_pool) +
+                len(self.incomp_pool),
+                fnum=len(self.fg_pool),
+                inum=len(self.incomp_pool),
+                bnum=len(self.bg_pool),
+                fr=self.fg_per_video,
+                br=self.bg_per_video,
+                ir=self.incomplete_per_video,
+                vnum=len(self.video_infos),
+                vc=self.video_centric,
+                stats=self.reg_stats))
+        else:
+            print("""
+            SSNDataset: proposal file {prop_file} parsed.
+            """.format(prop_file=self.proposal_file))
+
+    def load_annotations(self):
+        """Load annotation file to get video information."""
+        video_infos = []
+        if 'normalized_' in self.ann_file:
+            self.proposal_file = self.ann_file.replace('normalized_', '')
+            if not osp.exists(self.proposal_file):
+                print('{} does not exist. Converting from {}'.format(
+                    self.proposal_file, self.ann_file))
+                frame_dict = parse_directory(
+                    self.data_prefix, key_func=lambda x: x.split('/')[-1])
+                process_localize_proposal_list(self.ann_file,
+                                               self.proposal_file, frame_dict)
+                print('Finished conversion.')
+        else:
+            self.proposal_file = self.ann_file
+        proposal_infos = load_localize_proposal_file(self.proposal_file)
+        for p in proposal_infos:
+            if self.data_prefix is not None:
+                frame_dir = osp.join(self.data_prefix, p[0])
+            num_frames = int(p[1])
+            # gts:start, end, num_frames, class_label, tIoU=1
+            gts = [
+                SSNInstance(
+                    int(x[1]),
+                    int(x[2]),
+                    num_frames,
+                    label=int(x[0]),
+                    best_iou=1.0) for x in p[2] if int(x[2]) > int(x[1])
+            ]
```

Comment:
```python
gts = []
for x in proposal_info[2]:
    if int(x[2]) > int(x[1]):
        ssn_instance = SSNInstance(int(x[1]), int(x[2]), num_frames, label=int(x[0]), best_iou=1.0))
        gts.append(ssn_instance)
```
same to the following `proposals` var. btw, I think `int()` can be evitable. you can set it when generating `proposal_infos`.

Reply:
Now, code in ssn_utils is:
` proposal_boxes = [x.split() for x in video_info[offset: offset + num_proposals]]
`
If we want to set all parameters when generating `proposal_infos`. It will be:
```
for x in video_info[offset: offset + num_proposals]:
    info = x.split()
    info[0] = int(info[0])  
    info[1] = float(info[1])  
    info[2] = float(info[2])  
    info[3] = int(info[3])  
    info[4] = int(info[4])  
    proposal_boxes.append(info)  
```
Will it be better?

Reply:
Agree. And we can also add comment for each `info[x]` to explain its meaning.